File: app/src/main/java/com/android/messaging/data/db/ContactsDao.java
package com.android.messaging.data.db;

import java.util.List;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import com.android.messaging.data.model.Contact;

/**
 * Created by kalman.bencze on 02/11/2017.
 */
@Dao
public interface ContactsDao {
    @Query("SELECT * FROM Contact")
    List<Contact> getAll();

    @Insert
    void insertAll(Contact contacts);

    @Delete
    void delete(Contact contact);

    @Query("SELECT * FROM Contact "
            + "WHERE id = :contactId")
    List<Contact> getContact(int contactId);
}

File: app/src/main/java/com/android/messaging/data/db/ContactsDao.kt
package com.android.messaging.data.db

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import com.android.messaging.data.model.Contact
import org.intellij.lang.annotations.Language


/**
 * Created by kalman.bencze on 02/11/2017.
 */
@Dao
interface ContactsDao {
    @get:Query("SELECT * FROM Contacts")
    val all: List<Contact>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(contacts: List<Contact>)

    @Delete
    fun delete(contact: Contact)

    @Language("RoomSql")
    @Query("SELECT * FROM Contacts WHERE id = :contactId")
    fun getContact(contactId: Int): LiveData<Contact>

    @Language("RoomSql")
    @Query("SELECT * FROM Contacts ORDER BY id ASC")
    fun loadContacts(): LiveData<List<Contact>>
}
